spec/services/exports/export_distributions_csv_service_spec.rb

```diff
@@ -76,27 +75,76 @@
-
-        expect(subject[idx + 1]).to eq(row)
+    context 'while "Include in-kind value in donation and distribution exports?" is set to no' do
+      it 'should match the expected content without in-kind value of each item for the csv' do
+        expect(subject[0]).to eq(expected_headers)
+
+        distributions.zip(total_item_quantities).each_with_index do |(distribution, total_item_quantity), idx|
+          row = [
+            distribution.partner.name,
+            distribution.created_at.strftime("%m/%d/%Y"),
+            distribution.issued_at.strftime("%m/%d/%Y"),
+            distribution.storage_location.name,
+            distribution.line_items.where(item_id: item_id).total,
+            distribution.line_items.total_value.to_f,
+            distribution.delivery_method,
+            "$#{distribution.shipping_cost.to_f}",
+            distribution.state,
+            distribution.agency_rep,
+            distribution.comment
+          ]
+
+          row += total_item_quantity
+
+          expect(subject[idx + 1]).to eq(row)
+        end
+      end
+    end
+
+    context 'while "Include in-kind value in donation and distribution exports?" is set to yes' do
+      let(:expected_headers) { non_item_headers + all_org_items.pluck(:name).flat_map { |h| [h, "#{h} In-Kind Value"] } }
+
+      let(:quantities_and_values_of_items) do
+        template = all_org_items.pluck(:name).index_with({quantity: 0, value: Money.new(0), item_id: nil})
+
+        items_lists.map do |items_list|
+          row = template.deep_dup
+          items_list.each do |(item, quantity)|
+            row[item.name][:quantity] += quantity
+            row[item.name][:value] += Money.new(item.value_in_cents * quantity)
+            row[item.name][:item_id] = item.id
+          end
+          row.values
+        end
+      end
+
+      it 'should match the expected content with in-kind value of each item for the csv' do
+        allow(organization).to receive(:include_in_kind_values_in_exported_files).and_return(true)
+        expect(subject[0]).to eq(expected_headers)
+
+        distributions.zip(quantities_and_values_of_items).each_with_index do |(distribution, quantities_and_values_of_item), idx|
+          total_items_count = distribution.line_items.where(item_id: item_id).total
+          total_value = distribution.line_items.where(item_id: item_id).total_value.to_f
+          row = [
+            distribution.partner.name,
+            distribution.created_at.strftime("%m/%d/%Y"),
+            distribution.issued_at.strftime("%m/%d/%Y"),
+            distribution.storage_location.name,
+            total_items_count,
+            total_value,
+            distribution.delivery_method,
+            "$#{distribution.shipping_cost.to_f}",
+            distribution.state,
+            distribution.agency_rep,
+            distribution.comment
+          ]
+
+          row += quantities_and_values_of_item.flat_map { |item| [item[:quantity], item[:value].to_f] }
```

Comment:
We really should be setting up the data using explicit values rather than trying to calculate aggregations of random values. If it's not too much work, would you be able to make that change?